lenskit-api/src/main/java/org/lenskit/api/ItemBasedItemRecommender.java

```diff
@@ -29,62 +26,68 @@
+ * or People Also Liked feature.
  *
  * @see ItemRecommender
  * @author <a href="http://www.grouplens.org">GroupLens Research</a>
  * @compat Public
  * @since 0.10
  */
-public interface GlobalItemRecommender {
+public interface ItemBasedItemRecommender {
     /**
      * Recommend all possible items for a basket of items using the default exclude set.
      *
-     * @param items The items value.
+     * @param basket The reference basket.
      * @return The recommended items.
-     * @see #globalRecommend(Set, int, Set, Set)
+     * @see #recommendRelatedItems(Set, int, Set, Set)
      */
-    List<ScoredId> globalRecommend(Set<Long> items);
+    List<Long> recommendRelatedItems(Set<Long> basket);
 
     /**
      * Recommend up to <var>n</var> items for a basket of items using the default exclude set.
      *
-     * @param items The items value.
+     * @param basket The reference basket.
      * @param n     The number of recommendations to return.
      * @return The recommended items.
-     * @see #globalRecommend(Set, int, Set, Set)
+     * @see #recommendRelatedItems(Set, int, Set, Set)
      */
-    List<ScoredId> globalRecommend(Set<Long> items, int n);
+    List<Long> recommendRelatedItems(Set<Long> basket, int n);
 
     /**
-     * Recommend all possible items for a basket of items from a set of candidates using the default
-     * exclude set.
+     * Produce a set of recommendations for the item. This is the most general recommendation
+     * method, allowing the recommendations to be constrained by both a candidate set and an exclude
+     * set. The exclude set is applied to the candidate set, so the final effective candidate set is
+     * <var>candidates</var> minus <var>exclude</var>.
      *
-     * @param items      The items value.
-     * @param candidates The candidate set (can be null to represent the universe).
-     * @return The recommended items.
-     * @see #globalRecommend(Set, int, Set, Set)
+     * @param basket     The reference basket.
+     * @param n          The number of ratings to return. If negative, no specific size is requested.
+     * @param candidates A set of candidate items which can be recommended. If {@code null}, all
+     *                   items are considered candidates.
+     * @param exclude    A set of items to be excluded. If {@code null}, a default exclude set is
+     *                   used.
+     * @return A list of recommended items.
      */
-    List<ScoredId> globalRecommend(Set<Long> items, @Nullable Set<Long> candidates);
+    List<Long> recommendRelatedItems(Set<Long> basket, int n, @Nullable Set<Long> candidates,
+                                     @Nullable Set<Long> exclude);
 
     /**
-     * Produce a set of recommendations for the item. This is the most general recommendation
+     * Produce a set of recommendations for the item, with details. This is the most general recommendation
```

Comment:
Im not sure you get to have two functions say "this is the most general recommendation method" prehaps "this is the most powerful"?